human-viz-bridge/src/iCubEffortBridge.cpp

```diff
@@ -0,0 +1,340 @@
+    ITorqueControl *robotTorques;
+    vector<EffortPublisher> effortsList;
+
+public:
+    double getPeriod()
+    {
+        // module periodicity (seconds).
+        return 0.1;
+    }
+    // Main function. Will be called periodically every getPeriod() seconds
+    bool updateModule()
+    {
+        TickTime currentTime = normalizeSecNSec(yarp::os::Time::now());
+        
+        robotTorques->getTorques(m_robotConfiguration.data());
+        
+        for (size_t index = 0; index < effortsList.size(); ++index) {
+            EffortPublisher &effort = effortsList[index];
+            if (!effort.publisher) return false;
+            sensor_msgs_Temperature &effortMsg = effort.publisher->prepare();
+            effort.effortMsg.header.stamp = currentTime;
+            
+            double effort_temp = 0;
+            for (size_t indexJoint = 0; indexJoint < effortsList[index].jointEffort.nrDoFs; ++indexJoint) {
+                effort_temp += std::abs(m_robotConfiguration.getVal(index+indexJoint));
+            }
+            effort.effortMsg.temperature = effort_temp;
+            effortMsg = effort.effortMsg; 
+            effort.publisher->write();
+        }
+        
+        return true;
+    }
+    // Configure function. 
+    bool configure(ResourceFinder &rf)
+    {
+        m_robotConfigured = false;
+        
+        string moduleName = rf.check("name", Value("icub-effort-bridge"), "Checking module name").asString();
+        string urdfModelFile = rf.findFile("robotModelFilename");
+        setName(moduleName.c_str());
+        
+        //Robot model
+        const string robotModelFilename = rf.findFile("robotModelFilename");
+
+        vector<string> robot_jointList;
+        if (!parseStringListOption(rf.find("robot_JointList"), robot_jointList))
+        {
+            yError("Error while parsing the robot joint list");
+            return false;
+        }
+
+        iDynTree::ModelLoader modelLoader;
+              
+        if(!modelLoader.loadReducedModelFromFile(robotModelFilename, robot_jointList))
+        {
+            yError("Something wrong with the robot model loading!");
+            return false;
+        }
+    
+        iDynTree::Model robotModel = modelLoader.model(); 
+        iDynTree::Traversal traversal;
+        robotModel.computeFullTreeTraversal(traversal, robotModel.getDefaultBaseLink());
+        m_robotConfiguration.resize(robot_jointList.size());
+        m_robotConfiguration.zero();
+
+        size_t effortIndex = 0;
+        size_t DoFiterator = 1;
+        string robotJoint = "*";
+        vector<string> jointDegrees = {"_pitch", "_roll", "_yaw"};
+        for (size_t jointIndex = 0; jointIndex < robot_jointList.size(); jointIndex++) {
+            
+            effortsList.resize(effortIndex+1);
+            effortsList[effortIndex].jointEffort.nrDoFs = 0;
+            DoFiterator = 0;
+            
+            if (!(robot_jointList[jointIndex].find(robotJoint)!=string::npos)) {
```

Comment:
a `not` `not equal`, i.e. `! (... != ... )` is equivalent to `(... == ...)` which is also clearer